src/Features/Core/Portable/AddParameter/AbstractAddParameterCodeFixProvider.cs

```diff
@@ -113,125 +132,258 @@ internal abstract class AbstractAddParameterCodeFixProvider<
         }
 
         private int NonParamsParameterCount(IMethodSymbol method)
             => method.IsParams() ? method.Parameters.Length - 1 : method.Parameters.Length;
 
-        private async Task<Document> FixAsync(
+        private async Task<Solution> FixAsync(
             Document invocationDocument,
             IMethodSymbol method,
             TArgumentSyntax argument,
             SeparatedSyntaxList<TArgumentSyntax> argumentList,
+            bool fixAllReferences,
             CancellationToken cancellationToken)
         {
-            var methodDeclaration = await method.DeclaringSyntaxReferences[0].GetSyntaxAsync(cancellationToken).ConfigureAwait(false);
-
-            var (parameterSymbol, isNamedArgument) = await CreateParameterSymbolAsync(
-                invocationDocument, method, argument, cancellationToken).ConfigureAwait(false);
+            var solution = invocationDocument.Project.Solution;
+            var argumentType = await GetArgumentTypeAsync(invocationDocument, argument, cancellationToken).ConfigureAwait(false);
+            // the argumentNameSuggestion is the base for the parameter name. For each method declaration the name is made unique to avoid name collisions.
+            (var argumentNameSuggestion, var isNamedArgument) = await GetNameSuggestionForArgumentAsync(invocationDocument, argument, cancellationToken).ConfigureAwait(false);
+            var referencedSymbols = fixAllReferences
+                ? await FindMethodDeclarationReferences(invocationDocument, method, cancellationToken).ConfigureAwait(false)
+                : method.GetAllMethodSymbolsOfPartialParts();
+            // prepare lookup of declarations by document
+            var declarationLocations = referencedSymbols.SelectMany(definition
+                => definition.Locations.Select(location => new { Declaration = definition, Location = location })).ToImmutableArray();
+            // TODO: Insert hint in the fix with a warning if anySymbolsReferenceNotInSource is true
+            var anySymbolReferencesNotInSource = declarationLocations.Any(declarationLocation => !declarationLocation.Location.IsInSource);
+            var locationsInSource = declarationLocations.Where(declarationLocation => declarationLocation.Location.IsInSource);
+            var locationsByDocument = locationsInSource.ToLookup(declarationLocation
+                => solution.GetDocument(declarationLocation.Location.SourceTree));
+            foreach (var documentLookup in locationsByDocument)
+            {
+                var document = documentLookup.Key;
+                var syntaxFacts = document.GetLanguageService<ISyntaxFactsService>();
+                var syntaxRoot = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+                var editor = new SyntaxEditor(syntaxRoot, solution.Workspace);
+                var generator = editor.Generator;
+                foreach (var methodDeclaration in documentLookup)
```

Comment:
This looks quite good.  we def want to test that this feature works properly cross language.   Take a look at GenerateMethodCrossLanguageTests for how we do cross language testing.

Reply:
were cross language tests added?